### DISCUSSION TITLE
Trajectory vs Point queue mode

### DISCUSSION THREAD

**Discussion by olivierp99** [Yaskawa-Global/motoros2#243] [Q&A] [answered]
Hello,

I was wondering the difference when driving using trajectory mode vs point queue mode?

Thanks,

Olivier

**Comment by ted-miller** [comment, marked as answer]
Hi @olivierp99 

Due to memory constraints, the trajectory-mode has a limitation of 200 points per trajectory. There is a single message that comes to the robot which contains the entire trajectory.

The point-queue mode uses the exact same motion-processing as the trajectory mode. But it allows you to send an infinite number of messages which each contain only one point in the trajectory. 

Ultimately, the motion performance will be identical. But if you have a long trajectory (greater than 200 points), you should use the point-queue mode.

**Reply by gavanderhoorn** [reply]
Nit, but:

> Due to memory constraints, [..]

it's actually mostly protocol limitations at this point.

We could do longer trajectories.

**Reply by olivierp99** [reply]
Is there a buffer for the point-queue mode? Or my last point need to be processed before I send the new one? Or is the buffer 200 as well?

Thanks again

**Reply by ted-miller** [reply]
There is a buffer. New points should be submitted prior to the robot reaching the destination of the final point.

**Reply by gavanderhoorn** [reply]
I would perhaps describe it more as a *drip-feed for trajectories* with an on-controller buffer of `N` points instead of a streaming interface.

It's not a real-time motion interface -- which your last comment seems to hint at a little bit  @olivierp99.
